### DISCUSSION TITLE
[Q] Is spiral/jobs deprecated?

### DISCUSSION THREAD

**Discussion by fabis94** [roadrunner-server/roadrunner#787] [Q&A] [answered]
I can see that the `spiral/jobs` repo is archived: https://github.com/spiral/jobs

Does this mean that Jobs are no longer supported or maintained?

**Comment by rustatian** [comment, marked as answer]
Hey @fabis94 . https://github.com/spiral/jobs package works only with RR1. New RR2 contains an updated jobs plugin with all drivers supported by the old jobs. Thus, old https://github.com/spiral/jobs was deprecated in the flavor of new RR2 jobs.

**Reply by fabis94** [reply]
@rustatian Thank you! The documentation in this space is a bit lacking and it's even more confusing because it's often not clear if resources online are talking about RR1 or RR2. Can you point me in the right direction for getting started with Jobs with RR2? Specifically I'm interested in AMQP (RabbitMQ)

**Reply by rustatian** [reply]
Sure :) Here you are: https://roadrunner.dev/docs/beep-beep-jobs

**Reply by rustatian** [reply]
And the PHP part: https://github.com/spiral/roadrunner-jobs. Also, now you can declare/destroy pipelines in the runtime.

**Reply by fabis94** [reply]
@rustatian Thanks, I somehow missed that page! I see that Roadrunner works with tasks from a high abstraction level, abstracting away from the underlying implementation be it RabbitMQ, SQS or anything else. It also assumes that semantically the incoming messages will be "jobs" or expensive/time-consuming tasks.

My use case is that I want to use RabbitMQ as a replacement for HTTP requests entirely. Not just for expensive jobs, but for all incoming RPC requests between various microservices in an internal network. I also want to be able to work with AMQP directly - work with queues, exchanges, messages, transactions etc., so basically I want to work with the `php-amqplib` package.

I think the work you've done with Roadrunner is amazing, and I was hoping I could use it to get around the slow/blocking nature of PHP for my RabbitMQ server. It would definitely be better than just running message consumption in a `while` loop in a basic PHP script, that would only ever be able to process a single message concurrently.

Can my sort of custom use-case be achieved through Roadrunner? If it requires a bit of custom development, I'm not afraid of that.

Any tips and pointers appreciated, thanks in advance!

**Reply by rustatian** [reply]
> messages will be "jobs"

Yep, this is a high-level abstraction, but you can send any payload with it. You can populate only Job `ID` and Job `Payload` with your application payload. Also, in the new jobs plugin, we use our own implementation of the Priority queue, so, you can prioritize jobs between pipelines.

> AMQP directly - work with queues, exchanges, messages, transactions

RR2 allows you to define any exchange/queue/routing-key in the runtime. You may send an RPC `Declare` request to declare any queue in any exchange and then work with it. RR2 also Nack-s and Ack-s the `jobs` in case of error or if you send the protocol message to acknowledge even failed job (or made an attempt to retry if you want).

> I want to work with the php-amqplib package

Sorry, I'm not familiar with PHP, but I guess, that you may import this package to the worker and work with it. 

> Can my sort of custom use-case be achieved through Roadrunner?

Sure, if you send us a [FEATURE REQUEST] ticket and it would be possible to implement w/o breaking changes, I'll be happy to do that 😃 

> I think the work you've done with Roadrunner is amazing

Thank you very much 👍🏻
